src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -70,20 +70,73 @@ end
 ##### `cholesky`
 #####
 
-function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
-    F = cholesky(X)
-    function cholesky_pullback(Ȳ::Composite)
-        ∂X = if F.uplo === 'U'
-            chol_blocked_rev(Ȳ.U, F.U, 25, true)
-        else
-            chol_blocked_rev(Ȳ.L, F.L, 25, false)
-        end
-        return (NO_FIELDS, ∂X)
+function rrule(::typeof(cholesky), A::Real, uplo::Symbol=:U)
+    C = cholesky(A, uplo)
+    function cholesky_pullback(ΔC::Composite)
+        return NO_FIELDS, ΔC.factors[1, 1] / (2 * C.U[1, 1]), DoesNotExist()
     end
-    return F, cholesky_pullback
+    return C, cholesky_pullback
+end
+
+function rrule(
+    ::typeof(cholesky), A::Diagonal{<:Real}, ::Val{false}=Val(false); check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_pullback(ΔC::Composite)
+        check && !issuccess(C) && throw(PosDefException(C.info))
+        Ā = Diagonal(diag(ΔC.factors) .* inv.(2 .* C.factors.diag))
+        return NO_FIELDS, Ā, DoesNotExist()
+    end
+    return C, cholesky_pullback
+end
+
+# The appropriate cotangent is different depending upon whether A is Symmetric / Hermitian,
+# or just a StridedMatrix.
+# Implementation due to Seeger, Matthias, et al. "Auto-differentiating linear algebra."
+function rrule(
+    ::typeof(cholesky),
+    A::LinearAlgebra.HermOrSym{<:LinearAlgebra.BlasReal, <:StridedMatrix},
+    ::Val{false};
+    check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_pullback(ΔC::Composite)
+        Ā, U = _cholesky_pullback_shared_code(C, ΔC)
+        Ā = BLAS.trsm!('R', 'U', 'C', 'N', one(eltype(Ā)) / 2, U.data, Ā)
+        return NO_FIELDS, _symhermtype(A)(Ā), DoesNotExist()
+    end
+    return C, cholesky_pullback
+end
+
+function rrule(
+    ::typeof(cholesky),
+    A::StridedMatrix{<:LinearAlgebra.BlasReal},
+    ::Val{false}=Val(false);
+    check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_pullback(ΔC::Composite)
+        Ā, U = _cholesky_pullback_shared_code(C, ΔC)
+        Ā = BLAS.trsm!('R', 'U', 'C', 'N', one(eltype(Ā)), U.data, Ā)
+        idx = diagind(Ā)
+        @views Ā[idx] .= real.(Ā[idx]) ./ 2
+        return (NO_FIELDS, UpperTriangular(Ā), DoesNotExist())
+    end
+    return C, cholesky_pullback
+end
+
+function _cholesky_pullback_shared_code(C, ΔC)
+    issuccess(C) || throw(PosDefException(C.info))
+    U = C.U
+    Ū = ΔC.U
+    Ā = similar(U.data)
+    Ā = mul!(Ā, Ū, U')
+    Ā = LinearAlgebra.copytri!(Ā, 'U', true)
```

Comment:
I know complex matrices aren't officially supported in this PR, but I tested locally that this last fix makes them work for me for complex Hermitian matrices, for when the type constraints are relaxed.
```suggestion
    Ā = LinearAlgebra.copytri!(Ā, 'U', true)
    idx = diagind(Ā)
    @views Ā[idx] .= real(Ā[idx])
```

Reply:
Completely optional though, because some of the other `rrule`s wouldn't support complex either.